#!/usr/bin/env python
PACKAGE = "turtle_follower_ws"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 线速度 PID
gen.add("Kp_linear", double_t, 0, "线速度比例增益", 1.0, 0.0, 10.0)
gen.add("Ki_linear", double_t, 0, "线速度积分增益", 0.1, 0.0, 5.0)
gen.add("Kd_linear", double_t, 0, "线速度微分增益", 0.05, 0.0, 5.0)

# 角速度 PID
gen.add("Kp_angular", double_t, 0, "角速度比例增益", 4.0, 0.0, 20.0)
gen.add("Ki_angular", double_t, 0, "角速度积分增益", 0.2, 0.0, 10.0)
gen.add("Kd_angular", double_t, 0, "角速度微分增益", 0.1, 0.0, 10.0)

# 其它参数
gen.add("target_distance", double_t, 0, "目标跟随距离", 1.0, 0.0, 10.0)
gen.add("max_integral_linear", double_t, 0, "线速度积分限幅", 1.0, 0.0, 10.0)
gen.add("max_integral_angular", double_t, 0, "角速度积分限幅", 1.0, 0.0, 10.0)
gen.add("max_linear_speed", double_t, 0, "线速度最大值", 2.0, 0.0, 10.0)
gen.add("max_angular_speed", double_t, 0, "角速度最大值", 2.0, 0.0, 10.0)
gen.add("rate_hz", int_t, 0, "主循环频率(Hz)", 20, 1, 200)

exit(gen.generate(PACKAGE, "follower", "Follower"))